src/stepFunctions/commands/visualizeStateMachine/aslVisualization.ts

```diff
@@ -153,16 +153,28 @@ export class AslVisualization {
         )
 
         // When the panel is closed, dispose of any disposables/remove subscriptions
-        panel.onDidDispose(() => {
+        const disposePanel = () => {
             debouncedUpdate.cancel()
             this.onVisualizationDisposeEmitter.fire()
             this.disposables.forEach(disposable => {
                 disposable.dispose()
             })
             this.onVisualizationDisposeEmitter.dispose()
             this.isPanelDisposed = true
+        }
+
+        panel.onDidDispose(() => {
```

Comment:
onDidDispose returns a disposable, ensure it is handled.

Reply:
Disposable is pushed to this.disposables and is disposed of in disposePanel